versions/draft-3/SPEC.md

```diff
@@ -684,6 +690,34 @@ $task = 'task' $ws+ $identifier $ws* '{' $ws* $declaration* $task_sections $ws*
 
 For example, `task name { ... }`.  Inside the curly braces defines the sections.
 
+### Input Declarations
+
+Tasks declare inputs within the task block. For example:
+```wdl
+task t {
+  Int i
+  File f
+  Directory d
+  [...]
+}
+```
+
+#### Input Localization
+`File` and `Directory` inputs must be treated specially since they require localization to within the execution directory:
+- Files and directories are localized into the execution directory prior to the task execution commencing. 
+- When localizing a `File`, the engine may choose to place the file wherever it likes so long as it accords to these rules:
+  - The original file name must be preserved even if the path to it has changed.
+  - Two input files with the same name must be located separately, to avoid name collision.
+  - Two input files which originated in the same storage directory must also be localized into the same directory for task execution.
```

Comment:
This is a problem with DNAnexus cloud storage system, which allows two files to have the same name, and be placed in the same directory. This is allowed to support versioning.

Reply:
@orodeh Oh, huh, TIL! Out of interest, how do you handle that sort of localization collision today? It'll presumably make sense to codify this in such a way that you don't need to change anything

Reply:
Today, I keep the filename, but create a new local directory for them. For example,
assume file `A.txt` has two versions with ids `file-xxxx`, and `file-yyyy`.  If a task uses both files, then the first one will be localized as $HOME/A.txt, and the second to $HOME/file-yyyy/A.txt.

Reply:
This sounds like a good enough scheme to me. And it seems pretty unlikely to me that this would happen by accident without the author realizing so I don't think it would reduce portability. Out of interest how are you deciding which File is "first" in this case? 

How about this as a draft update:

```
#### Special Case: Versioning Filesystems
If two or more files in a versioning filesystem might have the same name in the same directory, the following special procedure can be used:
  - The first file is always placed as normal according to the usual rules.
  - Subsequent files that would otherwise overwrite this file are instead placed in a subdirectory named for the version.

For example imagine two versions of file fs://path/to/A.txt are being localized (labelled version 1.0 and 1.1). The first might be localized as /execution_dir/path/to/A. The second must then be placed in /execution_dir/path/to/1.1/A
```

Reply:
Sounds like a good solution to me. 

The dxWDL runtime serves the localized files on a first come first serve basis. The first file to be localized occupies `fs://path/to/A.txt`. All subsequent files with this name are placed in `fs://path/to/TAG/A.txt`.

Reply:
"can be used"

Is the use of the word "can" intentional here? It's a loaded term in standards docs